one_fm/tests/test_assign_pm_shift.py

```diff
@@ -0,0 +1,216 @@
+        for table, field in [
+            ("Shift Assignment", "employee"),
+            ("Shift Request", "employee"),
+            ("Employee", "name"),
+            ("Shift Type", "name"),
+            ("Operations Site", "name")
+        ]:
+            frappe.db.sql(f"DELETE FROM `tab{table}` WHERE {field} LIKE 'TEST_%'")
+        frappe.db.commit()
+
+    def test_roster_employee_creates_assignment(self):
+        """
+        Test: Verifies that a shift assignment is created for a roster employee.
+        """
+        roster = [frappe._dict({
+            "employee": self.roster_employee_name,
+            "employee_name": "John Doe",
+            "shift_type": self.shift_type_name,
+            "site": self.site_name,
+            "department": "Operations",
+            "roster_type": "Basic",
+            "shift": "PM Shift"
+        })]
+        create_shift_assignment(roster, self.test_date, "PM")
+        record_exists = frappe.db.exists("Shift Assignment", {"employee": self.roster_employee_name})
+        self.assertIsNotNone(record_exists)
+
+    def test_non_roster_employee_creates_assignment(self):
+        """
+        Test: Verifies that a shift assignment is created for a non-roster employee.
+        """
+        roster = [frappe._dict({
+            "employee": self.non_roster_employee_name,
+            "employee_name": "Jane Smith",
+            "shift_type": self.shift_type_name,
+            "site": self.site_name,
+            "department": "Engineering",
+            "roster_type": "Basic",
+            "shift": "PM Shift"
+        })]
+        create_shift_assignment(roster, self.test_date, "PM")
+        record_exists = frappe.db.exists("Shift Assignment", {"employee": self.non_roster_employee_name})
+        self.assertIsNotNone(record_exists)
+
+    def test_employee_on_leave_is_not_assigned(self):
+        """
+        Test: Verifies that no shift is created for an employee on leave (not present in roster).
+        """
+        roster = []  # Employee not present in roster
+        create_shift_assignment(roster, self.test_date, "PM")
+        record_exists = frappe.db.exists("Shift Assignment", {"employee": self.roster_employee_name})
+        self.assertIsNone(record_exists)
+
+    def test_create_shift_assignment_with_shift_request(self):
+        """
+        Test: create_shift_assignment creates assignment for employee with approved shift request.
+        """
+        roster = [frappe._dict({
+            "employee": self.roster_employee_name,
+            "employee_name": "John Doe",
+            "shift_type": self.shift_type_name,
+            "site": self.site_name,
+            "department": "Operations",
+            "roster_type": "Basic",
+            "shift": "PM Shift"
+        })]
+        create_shift_assignment(roster, self.test_date, "PM")
+        record_exists = frappe.db.exists("Shift Assignment", {"employee": self.roster_employee_name})
+        self.assertIsNotNone(record_exists)
```

Comment:
@samdanikouser this assertion is the same as in test_roster_employee_creates_assignment, are we not meant to check the shift details ?